lib/history/memory/inmemory_test.go

```diff
@@ -41,57 +42,62 @@ func (s *InMemorySuite) SetUpSuite(c *C) {
 	s.clock = clockwork.NewFakeClock()
 }
 
-// TestRecordStatus simply tests that the timeline can successfully record
-// a status.
-func (s *InMemorySuite) TestRecordStatus(c *C) {
-	test.WithTimeout(func(ctx context.Context) {
-		timeline := s.newTimeline()
-		node := "test-node"
-		old := &pb.NodeStatus{Name: node, Status: pb.NodeStatus_Running}
-		expected := []*pb.TimelineEvent{history.NewNodeRecovered(s.clock.Now(), node)}
+// TestRecordEvents verifies that the timeline can successfully record a new
+// event.
+func (s *InMemorySuite) TestRecordEvents(c *C) {
+	timeline := s.newTimeline()
+	events := []*pb.TimelineEvent{pb.NewNodeHealthy(s.clock.Now(), node)}
+	expected := []*pb.TimelineEvent{pb.NewNodeHealthy(s.clock.Now(), node)}
+	comment := Commentf("Expected the a node recovered event to be recorded.")
 
-		c.Assert(timeline.RecordStatus(ctx, old), IsNil)
+	test.WithTimeout(func(ctx context.Context) {
+		c.Assert(timeline.RecordEvents(ctx, events), IsNil, comment)
 
 		actual, err := timeline.GetEvents(ctx, nil)
-		c.Assert(err, IsNil)
-		c.Assert(actual, test.DeepCompare, expected, Commentf("Expected the status to be recorded."))
+		c.Assert(err, IsNil, comment)
+		c.Assert(actual, test.DeepCompare, expected, comment)
 	})
 }
 
 // TestFIFOEviction tests the FIFO eviction policy of the timeline.
 func (s *InMemorySuite) TestFIFOEviction(c *C) {
+	// Limit timeline capacity to 1.
+	// Necessary to easily test if events are evicted if max capacity is reached.
+	timeline := s.newLimitedTimeline(1)
+	events := []*pb.TimelineEvent{
+		pb.NewNodeHealthy(s.clock.Now().Add(-time.Second), node), // older event should be evicted first
+		pb.NewNodeDegraded(s.clock.Now(), node),
+	}
+	expected := []*pb.TimelineEvent{pb.NewNodeDegraded(s.clock.Now(), node)}
+	comment := Commentf("Expected node degraded event.")
+
 	test.WithTimeout(func(ctx context.Context) {
-		// Limit timeline capacity to 1.
-		// Necessary to easily test if events are evicted if max capacity is reached.
-		timeline := s.newLimitedTimeline(1)
-		node := "test-node"
-		old := &pb.NodeStatus{Name: node, Status: pb.NodeStatus_Running}
-		new := &pb.NodeStatus{Name: node, Status: pb.NodeStatus_Degraded}
-		expected := []*pb.TimelineEvent{history.NewNodeDegraded(s.clock.Now(), node)}
-
-		// Recording two statuses should result in timeline to exceed capacity.
-		c.Assert(timeline.RecordStatus(ctx, old), IsNil)
-		c.Assert(timeline.RecordStatus(ctx, new), IsNil)
+
+		// Recording two statuses should result in timeline exceeding capacity.
+		c.Assert(timeline.RecordEvents(ctx, events), IsNil, comment)
+
+		fmt.Println(timeline.events)
```

Comment:
This looks like debugging aid - if it's supposed to be in the test, I'd use the provided `c.Logf` as it can be controlled via command line.